src/lib/fsm/index.ts

```diff
@@ -0,0 +1,54 @@
+export default class fsm {
+  private currentState: string;
+  private states: { [id: string]: { [id: string]: string } } = {};
+
+  constructor(initialState: string, transitions: Array<transition>) {
+    this.currentState = initialState;
+
+    transitions.forEach((transition) => this.addTransition(transition));
+  }
+
+  public CurrentState(): string {
+    get: {
+      return this.currentState;
+    }
+  }
+
+  public handle(message: string) {
+    if (this.states[this.currentState] != undefined) {
+      var target = this.states[this.currentState][message];
+
+      if (target != undefined) {
+        this.currentState = target;
+      }
+    }
+  }
+
+  public static Transition(fromState: string, message: string, toState: string): transition {
+    return new transition(fromState, message, toState);
+  }
```

Comment:
Transition to it's own file.
Naming convention: method should be lower case
Naming convention: class name should be pascal case.